Documentation/pinning.md

```diff
@@ -0,0 +1,83 @@
+# Object pinning
+
+BPF has a persistent view of maps and programs under its own filesystem
+`/sys/fs/bpf`. Users are able to make each object visible under the bpffs.
+We call it `object pinning`. This is done by calling syscall `bpf(2)` with
+a command `BPF_OBJ_PIN`. After doing that, users are able to use the object
+with commands such as `BPF_OBJ_GET`, or remove the object with an ordinary
+VFS syscall `unlink(2)`.
+
+Doing that, we can make maps and programs stay alive across process
+terminations. This mechanism provides a much more consistent way of sharing
+objects with other processes, compared to other solutions such as `tc`,
+where objects are shared via Unix domain sockets.
+
+## Different pinning options
+
+`C.bpf_map_def.pinning` (defined in
+[bpf.h](https://github.com/iovisor/gobpf/blob/master/elf/include/bpf.h#L616))
```

Comment:
We shouldn't link to the master branch, line numbers won't match if the code is updated.

Reply:
@iaguis What do you suggest? just remove the link and write `(defined in elf/include/bpf.h)`?

Reply:
Or use https://github.com/iovisor/gobpf/blob/17d86e402/elf/include/bpf.h#L616

Reply:
@iaguis Yeah, I realized that the line number would not match in the future. But using a specific commit would be also a broken approach, if we force push. I'll try to update it with a specific commit this time, and afterwards I'll try not to force push again in this PR. :-)

Reply:
I think you can merge after this and hopefully nobody force-pushes to master :)

Reply:
@iaguis Done.